### ISSUE TITLE
BibKnowledge: get_kb_mappings_json not to return key if no value

### ISSUE THREAD

**Issue by jeromecaffaro** [inveniosoftware/invenio#1044]
Originally on 2012-05-08

Currently when using `get_kb_mappings_json()` for a key that has no value, the key is returned as value. For example:

```
 foobarKB:
    foo1 --> bar1
    foo2 -->
    foo3 --> bar3

  get_kb_mappings_json('foobarKB', 'foo2')
  {'label': 'foo2', 'value': 'foo2'}
```
source:modules/bibknowledge/lib/bibknowledge.py@a6221e6#L113

This is making impossible to disambiguate between the two cases:
```
foobarKB1:                      |            foobarKB2:
    foo1 --> bar1               |                foo1 --> bar1
    foo2 -->                    |                foo2 --> foo2
    foo3 --> bar3               |                foo3 --> bar3
```

One could be willing to receive `null` or empty string instead:
```
get_kb_mappings_json('foobarKB', 'foo2')
>>{'label': 'foo2', 'value': null}
```

or 

```
get_kb_mappings_json('foobarKB', 'foo2')
>>{'label': 'foo2', 'value': ""}
```

This function interface could also be extended similarly to [source:modules/bibformat/doc/hacking/bibformat-api.webdoc@1bb7957#L334 bibformat_engine.BibFormatObject.kb(kb, string, default="")] so that the default returned value can be decided:
```
get_kb_mappings_json('foobarKB', 'foo2', default="")
>>{'label': 'foo2', 'value': ""
get_kb_mappings_json('foobarKB', 'foo2', default='foo2')
>>{'label': 'foo2', 'value': 'foo2'}}
```

This might extend to other kb-related functions, and REST APIs.

(As briefly mused with Joe about one year ago, there is no reason not to move to this new interface)

**Comment by jirikuncar** [comment]
@hachreak can you confirm that it has been solved in 2.0?

**Comment by hachreak** [comment]
In REST API we can choice if it should return:
 1. list of keys only
 2. list of values only
 3. list of key/values

This is an example of knowledge `Faq`:

![screenshot - 03272015 - 10 18 30 am](https://cloud.githubusercontent.com/assets/377047/6865290/95bb0370-d46c-11e4-8f5b-d7566aa6be2d.png)


1. Returns keys:

```bash
$> http GET http://0.0.0.0:8080/api/knowledge/faq/mappings/from
HTTP/1.0 200 OK
Content-Length: 26
Content-Type: application/json
Date: Fri, 27 Mar 2015 09:23:43 GMT
Server: Werkzeug/0.10.1 Python/2.7.6

[
    "banana", 
    "foo", 
    "apple"
]
```

2. Returns values: in this case return a empty string. 
If necessary, we can add an argument to ignore empty values.

```bash
$> http GET http://0.0.0.0:8080/api/knowledge/faq/mappings/to
HTTP/1.0 200 OK
Content-Length: 20
Content-Type: application/json
Date: Fri, 27 Mar 2015 09:24:43 GMT
Server: Werkzeug/0.10.1 Python/2.7.6

[
    "", 
    "bar", 
    "fruit"
]
```

3. Returns key/value pairs: if value is not defined, it returns an empty string.

```bash
$> http GET http://0.0.0.0:8080/api/knowledge/faq
HTTP/1.0 200 OK
Content-Length: 94
Content-Type: application/json
Date: Fri, 27 Mar 2015 09:23:07 GMT
Server: Werkzeug/0.10.1 Python/2.7.6

[
    {
        "from": "banana", 
        "to": ""
    }, 
    {
        "from": "foo", 
        "to": "bar"
    }, 
    {
        "from": "apple", 
        "to": "fruit"
    }
]
```
